idaes/core/base/costing_base.py

```diff
@@ -337,13 +337,19 @@ def register_flow_type(self, flow_type, cost):
         """
         self.flow_types.add(flow_type)
 
-        # Create a Var to hold the cost
-        # Units will be different between flows, so have to use scalar Vars
-        fvar = pyo.Var(
-            units=pyo.units.get_units(cost), doc=f"Cost parameter for {flow_type} flow"
-        )
-        self.add_component(f"{flow_type}_cost", fvar)
-        fvar.fix(cost)
+        name = f"{flow_type}_cost"
+        current_component = self.component(name)
+        if current_component is not None and current_component is cost:
+            pass
+        else:
```

Comment:
I think I see an edge case here that needs to be caught - what happens if a component `name` already exists but is not `cost`? I think this would implicitly replace the existing component, so I think we need to catch this and raise an `Exception`.

Reply:
I agree -- I was just making a minimal change to existing behavior. Happy to add an `Exception` here though.

Reply:
I think an `Exception` is the best approach - it is easy to add in here, and I generally never try to second guess the user. I.e. if they do something wrong it is best to tell them to fix it rather than make assumptions.